### ISSUE TITLE
data okta_network_zone fails to retrieve information

### ISSUE THREAD

**Issue by JBorregoE** [okta/terraform-provider-okta#1994]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.8.2
on darwin_amd64
+ provider registry.terraform.io/okta/okta v4.8.1

### Affected Resource(s)

- data okta_network_zone

### Terraform Configuration Files

```hcl
data okta_network_zone barcelona_netzone {
  name = "Barcelona"
}
```

### Debug Output

╷
│ Error: failed to find network zone: json: cannot unmarshal object into Go struct field NetworkZone.locations of type []*sdk.NetworkZoneLocation
│ 
│   with data.okta_network_zone.barcelona_netzone,
│   on okta-network_zones.tf line 43, in data "okta_network_zone" "barcelona_netzone":
│   43: data okta_network_zone barcelona_netzone {
│ 
╵

### Expected Behavior

`data okta_network_zone` retrieves the information from the existing zones without any issue.

#### Can this be done in the Admin UI?

I don't know

#### Can this be done in the actual API call?

I don't know

### Actual Behavior

data okta_network_zones produces an error.

### Steps to Reproduce

1. Define a `data okta_network_zone` resource that exists in Okta.
2. `terraform plan` or `terraform apply`
3. The resource is not reachable.

### Important Factoids

We run a task daily to monitor changes on Okta, and this morning failed for the first time. However, there were no recent changes in our environment.

**Comment by JBorregoE** [comment]
We reorganised our repository and imported the zones into the state. Using the resource `okta_network_zone` instead of the data block did not fail.
If this issue is not causing troubles for anyone else can be closed. Otherwise, happy to provide extra information if needed.

**Comment by Abajaj-dh** [comment]
I am also experiencing the same issue. There is no change in the repository or version. The data object still works with id but not with the name.

**Comment by duytiennguyen-okta** [comment]
@Abajaj-dh do you have the log when you run `TF_LOG=debug terraform apply`?

**Comment by Abajaj-dh** [comment]
Hi @duytiennguyen-okta - yes, I've exported the logs. But it's quite huge. I've shared the specific log related to network zone. 
[ErrorLog1.txt](https://github.com/user-attachments/files/16391869/ErrorLog1.txt)

**Comment by duytiennguyen-okta** [comment]
It should be fixed with #2057

**Comment by Abajaj-dh** [comment]
Thank you for your response and fix. It's working now :)
